OMCompiler/Compiler/Template/CodegenFMUCommon.tpl

```diff
@@ -99,6 +99,18 @@ match simCode
 case SIMCODE(modelInfo=modelInfo) then
 match modelInfo
 case MODELINFO(vars=SIMVARS(stateVars=stateVars)) then
+  if isFMIVersion20(FMUVersion) then
+  <<
+  <ModelVariables>
+  <%System.tmpTickReset(0)%>
+  <%vars.modelDescriptionVars |> var =>
```

Comment:
This should probably loop all other var lists (since I would like to avoid the duplicates in `modelDescriptionVars`) and only dump variables if `fmi_variable_index != NONE()`.

Reply:
It will not loop all var list, this part is moved to simcode phase as we collect all the variable there
see function "protected function createFmiModelDescriptionVariable" in SimCodeUtil.mo